substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/meta/AnalysisMetaAccess.java

```diff
@@ -47,6 +48,14 @@ public AnalysisType lookupJavaType(Class<?> clazz) {
         return (AnalysisType) super.lookupJavaType(clazz);
     }
 
+    public AnalysisType lookupJavaType(String className) {
```

Comment:
I agree. I was very unhappy making the commit https://github.com/oracle/graal/pull/3523/commits/39374fc5a798c01a4b5174b24a3fb2503003f067
There is actually only one place where the method is used - the  [PermissionsFeature](https://github.com/oracle/graal/blob/39374fc5a798c01a4b5174b24a3fb2503003f067/substratevm/src/com.oracle.svm.truffle.tck/src/com/oracle/svm/truffle/tck/PermissionsFeature.java) in the *com.oracle.svm.truffle.tck* project. Unfortunately, the CheckStyle configuration of that project does not allow dynamic class loading. I was not sure whether I'm allowed to change the CheckStyle config, but if you think it is reasonable, I can do so. Or alternatively, move the utility method somewhere else. However, since there is only one use of it, I would be happier if it could stay in  *com.oracle.svm.truffle.tck*

Reply:
I moved the class loading back to the [PermissionsFeature](https://github.com/oracle/graal/blob/39374fc5a798c01a4b5174b24a3fb2503003f067/substratevm/src/com.oracle.svm.truffle.tck/src/com/oracle/svm/truffle/tck/PermissionsFeature.java).